### PR TITLE
add librabry dependencies

### PR DESCRIPTION
**Pull request by PolyMorpher01**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,28 +1,54 @@
...
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
-    }
-    buildTypes {
-        release {
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-        }
+  compileSdkVersion 28
+  defaultConfig {
+    applicationId "com.example.ayush.weatherapp"
```
Comment (subash-poudel) [inline app/build.gradle]: com.example is reserved packagename and is not suitable. change to com.ayush.weatherapp

File: app/build.gradle [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
+    minSdkVersion 25
```
Comment (subash-poudel) [inline app/build.gradle]: minsdkversion is too high

File: app/build.gradle [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation 'com.android.support:appcompat-v7:28.0.0-rc01'
-    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+  implementation fileTree(dir: 'libs', include: ['*.jar'])
+  implementation 'com.android.support:appcompat-v7:28.0.0-rc01'
+  implementation 'com.android.support.constraint:constraint-layout:1.1.2'
+
```
Comment (subash-poudel) [inline app/build.gradle]: multiple lines

File: app/build.gradle [hunk 4]
```diff
@@ -1,28 +1,54 @@
...
+  debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.6.1'
+
+  implementation 'com.squareup.retrofit2:retrofit:2.4.0'
+
+  implementation 'io.reactivex.rxjava2:rxandroid:2.0.2'
+
+  implementation 'com.google.dagger:dagger:2.16'
+  annotationProcessor 'com.google.dagger:dagger-compiler:2.16'
+
+  implementation 'com.facebook.stetho:stetho:1.3.1'
```
Comment (subash-poudel) [inline app/build.gradle]: older version

File: app/build.gradle [hunk 5]
```diff
@@ -1,28 +1,54 @@
...
+  implementation 'com.jakewharton:butterknife:8.8.1'
+  annotationProcessor 'com.jakewharton:butterknife-compiler:8.8.1'
+
+  implementation 'com.jakewharton.timber:timber:4.7.1'
+
+  debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.6.1'
+
+  implementation 'com.squareup.retrofit2:retrofit:2.4.0'
+
+  implementation 'io.reactivex.rxjava2:rxandroid:2.0.2'
```
Comment (subash-poudel) [inline app/build.gradle]: rxjava ??

File: app/build.gradle (hunk 1 above)
Comment (PolyMorpher01) [reply app/build.gradle]: :ok:

File: app/build.gradle (hunk 2 above)
Comment (PolyMorpher01) [reply app/build.gradle]: :ok:

File: app/build.gradle (hunk 3 above)
Comment (PolyMorpher01) [reply app/build.gradle]: :ok:

File: app/build.gradle (hunk 5 above)
Comment (PolyMorpher01) [reply app/build.gradle]: :ok:

File: app/build.gradle (hunk 4 above)
Comment (PolyMorpher01) [reply app/build.gradle]: :ok:

**Pull request merged**